middleware/basic_auth_test.go

```diff
@@ -11,11 +12,139 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestBasicAuthWithConfig(t *testing.T) {
+	validatorFunc := func(u, p string, c echo.Context) (bool, error) {
+		if u == "joe" && p == "secret" {
+			return true, nil
+		}
+		if u == "error" {
+			return false, errors.New(p)
+		}
+		return false, nil
+	}
+	defaultConfig := BasicAuthConfig{Validator: validatorFunc}
+
+	// we can not add OK value here because ranging over map returns random order. We just try to trigger break
+	tooManyAuths := make([]string, 0)
+	for i := 0; i < extractorLimit+2; i++ {
+		tooManyAuths = append(tooManyAuths, basic+" "+base64.StdEncoding.EncodeToString([]byte("nope:nope")))
+	}
+
+	var testCases = []struct {
+		name         string
+		givenConfig  BasicAuthConfig
+		whenAuth     []string
+		expectHeader string
+		expectErr    string
+	}{
```

Comment:
Suggestion:
- wantHeader
- wantErr

Reply:
I am leaving this as is because we have not used `want*` in a single test in this repo. `expect*` is the convention.